src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList() {
+        size = 0;
+        elements = new Object[START_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (elements.length == size) {
+            growCapacity();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexValidationForAdd(index);
+        if (elements.length == size) {
+            growCapacity();
+        }
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexValidation(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidation(index);
+        T deleted = (T) elements[index];
+        System.arraycopy(elements, index + 1, elements, index, size - index - 1);
+        size--;
+        return deleted;
     }
 
     @Override
     public T remove(T t) {
+        for (int index = 0; index < size; index++) {
+            if ((t == null || elements[index] == null)
+                    ? elements[index] == t : elements[index].equals(t)) {
+                return remove(index);
+            }
+        }
+        if (true) {
+            throw new NoSuchElementException("The element is not found");
+        }
         return null;
```

Comment:
```suggestion
        throw new NoSuchElementException("The element is not found");
```
if you throw an Exception - the program stops working 
you don't need line 78 because you will never get there
also, `if (true)` makes no sense, right?

Reply:
That's right, thank you. Fixed.